### DISCUSSION TITLE
List of still needed models

### DISCUSSION THREAD

**Discussion by jansenbe** [pnp/pnpcore#188] [General] [answered]
Lets collect a list of things we still need for GA and identify which of these are good for new contributors joining the team.

Based upon our chat this list came up (not in any particular order):

- ALM 
- Teams model completion + test cases
- SharePoint Permissions 
- Navigation
- Hub site joining
- Sharing 
- Search
- Resources (for multi-lingual needs)
- List view handling

Tagging @ypcode , @pkbullock , @PaoloPia and @JarbasHorst

**Comment by ypcode** [comment]
I don't see big things lacking for common modern SharePoint development..

 - SP Webhooks subscriptions (Should we also already start thinking of MS Graph subscriptions... I am actually not sure if they are feature equal to SP subscriptions).
- Changes API

Other than this,

- RegionalSettings
- Alerts
- Attachments
- Document Sets
- CheckedOutFiles

I am not sure if SP Workflow is still worthy since it's now being preferably built on Power Automate (If I got it well)

**Comment by pkbullock** [comment]
I'm thinking:

- List/Column formatting (satisfy those war horses)
- Site Headers
- Navigation as above, including Hub, Site and Footer
- Site Creation
- Pages (already listed in Issues list)

**Reply by ypcode** [reply]
(I was thinking site creation and list formatting were already implemented :D) Great addition ! :)

**Comment by pkbullock** [comment]
I am seeing quite a few TODOs within the source in the models, these may need to be included.

**Comment by pkbullock** [comment]
OK up to this point updated the issues list with the above. The Resources think Paolo took care of, and I have already done some on the list views.  So this is a line in the sand for the first batch.

**Comment by jansenbe** [comment, marked as answer]
We're tracking the still needed models via the issue list, so marking this one as answered to keep thing clean
